### DISCUSSION TITLE
"ignore" crate: Possible to test single paths for ignore status?

### DISCUSSION THREAD

**Discussion by blinry** [BurntSushi/ripgrep#2862] [Q&A] [answered]
## Use case

We're using the [ignore](https://crates.io/crates/ignore) crate in [our project](https://github.com/ethersync/ethersync), and would like to check for single paths whether or not they match any of the default ignore rules.

Is that currently possible?

## Example

We're thinking of something like using `Walk`'s `skip_entry()` directly. Here with a fictional `IgnoreTestBuilder`:

```rust
let tester = IgnoreTestBuilder.new("the/directory").standard_filters(true);
assert!(tester.test_ignore("tmp/log_output"), true);
```

## Current workaround

We're going through the entire `Walk` iterator, and then check if our path is in it.

**Comment by BurntSushi** [comment, marked as answer]
No, it isn't. This is something I'd like to support in the future, but the entire `ignore` crate needs to be re-thought from the ground-up before I'd be willing to tackle it. I've been saying for a long time that the crate needs to be rewritten and it hasn't happened yet, so I wouldn't hold my breath unfortunately.

If this is a requirement for your project, you'll need to implement this on your own. You can probably reuse some components from `ignore` though, like the `Gitignore` matcher.

**Reply by blinry** [reply]
Thanks for the quick reply!
